SAPex/Controllers/SandboxesController.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BusinessLogicLayer.DtoModels;
+using BusinessLogicLayer.Interfaces;
+using BusinessLogicLayer.Services;
+using DataAccessLayer.Service;
+using DbMigrations.Data;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+using PagedList;
+using SAPex.Mappers;
+using SAPex.Models;
+using SAPex.Models.Validators;
+
+namespace SAPex.Controllers
+{
+    [Route("api/sandboxes")]
+    [ApiController]
+    public class SandboxesController : ControllerBase
+    {
+        private readonly ISandboxService _sandboxService;
+        private readonly IStackTechnologyService _stackTechnologyService;
+        private readonly ILanguageService _languageService;
+        private readonly SandboxMapper _mapper;
+
+        public SandboxesController(ISandboxService sandboxService, IStackTechnologyService stackTechnologyService, ILanguageService languageService)
+        {
+            _sandboxService = sandboxService;
+            _stackTechnologyService = stackTechnologyService;
+            _languageService = languageService;
+            _mapper = new SandboxMapper();
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get([FromRoute] Guid id)
+        {
+            SandboxDtoModel sandboxDtoModel = await _sandboxService.FindSandboxByIdAsync(id); // on what level goes null check? it should go directly to db?
+            IEnumerable<StackTechnologyDtoModel> stackTechnologyDtoModel = await _stackTechnologyService.GetStackTechnologiesBySandboxIdAsync(id);
+            IEnumerable<LanguageDtoModel> languageDtoModel = await _languageService.GetLanguagesBySandboxIdAsync(id);
+
+            if (sandboxDtoModel == null)
+            {
+                return await Task.FromResult(NotFound());
+            }
+
+            SandboxViewModel viewModel = _mapper.MapSbStackLgFromDtoToView(sandboxDtoModel, languageDtoModel, stackTechnologyDtoModel);
+
+            return await Task.FromResult(Ok(viewModel)); // convert to json?
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetByPage(int pageNumber = 1, int pageSize = 5, string sortOrder = "name", string searchString = "")
```

Comment:
Please use one ViewModel with fields as an input parameter.

Reply:
have little problem with passing parameter to method